File: cpp/32c_1619A_SquareString.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, flag;
    std::vector<std::string> ans;
    std::string temp;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>temp;
        n = temp.size();
        if(n % 2 != 0){
            ans.push_back("NO");
        }
        else{
            flag = 0;
            for(int j = 0; j < (n / 2); j++){
                if(temp[j] != temp[(n / 2) + j]){
                    ans.push_back("NO");
                    flag = 1;
                    break;
                }
            }
            if (flag == 0){
                ans.push_back("YES");
            }
        }
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/32c_1619A_SquareString.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        temp = input().strip()
        n = len(temp)
        temp1, temp2 = temp[:(n//2)], temp[(n//2):]
        if temp1 == temp2:
            ans.append("YES")
        else:
            ans.append("NO")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()